src/SourceIndexServer/Models/ResultsHtmlGenerator.cs

```diff
@@ -208,9 +209,13 @@ private void AppendAffiliateLinks(Query query)
             var term = query.OriginalString;
             term = Markup.UrlEncodeAndHtmlEscape(term);
             WriteLine("<ul>");
-            AppendAffiliateLink("http://www.bing.com/search?q=" + term);
-            AppendAffiliateLink("http://social.msdn.microsoft.com/Search/en-US?query=" + term);
-            AppendAffiliateLink("http://stackoverflow.com/search?q=" + term);
+            // Read the AffiliateLinks file and display links one by one.
+            
+            foreach (string line in File.ReadLines("..\\AffiliateLinks.txt"))

```

Comment:
Could you please surround this with if (File.Exists(...)) { foreach ... }, such that if the file is not found for some reason we just continue and not crash in a horrible way?